packages/mobile/src/send/Send.tsx

```diff
@@ -194,20 +214,43 @@ class Send extends React.Component<Props, State> {
     navigate(Screens.SendAmount, { recipient })
   }
 
-  render() {
-    const { t, defaultCountryCode } = this.props
-    const { loading, searchQuery, recentFiltered, allFiltered } = this.state
+  onPermissionsAccepted = async () => {
+    this.props.importContacts()
+
+    const { recipientCache } = this.props
+
+    if (recipientCache) {
```

Comment:
The recip cache at this point will be an empty object. The `importContacts` call above triggers the saga, which will take some time. Can you clarify what this block is intended to do?

Reply:
Ah I thought that it was updated immediately, but now I see that it doesn't, and all that is needed here is to reset the searchQuery, hasGivenpermission, and recentRecipients (this last one is needed otherwise the Celo Default Recipient gets added again when reloaded. I will remove the unnecessary code